src/index.rs

```diff
@@ -260,27 +262,40 @@ impl std::error::Error for OutOfBoundsError {}
 
 /// Indicates that the `Token` could not be parsed as valid RFC 6901 index.
 #[derive(Debug, PartialEq, Eq)]
-pub struct ParseIndexError {
-    /// The source `ParseIntError`
-    pub source: ParseIntError,
+pub enum ParseIndexError {
+    /// The Token does not represent a valid integer.
+    InvalidInteger(ParseIntError),
+    /// The Token contains leading zeros.
+    LeadingZeros,
```

Comment:
This is unfortunately a breaking change, but I don't see another way to represent the additional error condition.

Reply:
Yea, that needs to go in.